src/model/delta/insertdelta.js

```diff
@@ -95,7 +104,44 @@ register( 'insert', function( position, nodes ) {
 	delta.addOperation( insert );
 	this.document.applyOperation( insert );
 
+	// When element is a DocumentFragment we need to move its markers to Document#markers.
+	if ( nodes instanceof DocumentFragment ) {
+		for ( const marker of nodes.markers ) {
+			const doc = this.document;
+			const range = moveRange( marker[ 1 ], position );
+			const markerDelta = new MarkerDelta();
+			const markerOperation = new MarkerOperation( marker[ 0 ], null, range, doc.markers, doc.version );
+
+			this.addDelta( markerDelta );
+			markerDelta.addOperation( markerOperation );
+			doc.applyOperation( markerOperation );
+		}
+	}
+
 	return this;
 } );
 
 DeltaFactory.register( InsertDelta );
+
+// Moves range relative to given position.
+//
+// @param {module:engine/model/range~Range} range
+// @param {module:engine/model/position~Position} position
+// @returns {module:engine/model/range~Range} Moved range.
+function moveRange( range, position ) {
+	const positionRoot = position.parent.root;
+
+	// Clone paths.
+	let startPath = range.start.path.concat( [] );
+	let endPath = range.end.path.concat( [] );
+
+	// Move range out of DocumentFragment.
+	startPath.shift();
+	endPath.shift();
+
+	// Set Range on position.
+	startPath = position.path.concat( startPath );
+	endPath = position.path.concat( endPath );
```

Comment:
I expect the same but:
- position where I want to insert element is `<root>abc[]</root>` -> path is `[ 3 ]`
- start position of my range in doc frag `<docfrag><p>d[]ef<p></docfrag>` `[ 0, 1 ]`

When I create position in this way `new Position( root, marker.start.path )` the result path will be `[ 0, 1 ]` but I need `[ 3, 0, 1 ]`.